File: spatial-index/src/test/java/lva/spatialindex/index/RectanglesTest.java
package lva.spatialindex.index;

import org.junit.Test;

import java.awt.Rectangle;

import static org.junit.Assert.assertEquals;

/**
 * @author vlitvinenko
 */
public class RectanglesTest {
    @Test
    public void test_area() {
        assertEquals(0, Rectangles.area(new Rectangle()));
        assertEquals(8, Rectangles.area(new Rectangle(0, 0, 2, 4)));
    }

    @Test
    public void test_margin() {
        assertEquals(0, Rectangles.margin(new Rectangle()));
        assertEquals(12, Rectangles.margin(new Rectangle(0, 0, 2, 4)));
    }
}

File: spatial-index/src/test/kotlin/lva/spatialindex/index/RectanglesTest.kt
package lva.spatialindex.index

import java.awt.Rectangle
import kotlin.test.Test
import kotlin.test.assertEquals

/**
 * @author vlitvinenko
 */
class RectanglesTest {
    @Test
    fun test_area() {
        assertEquals(0, Rectangle(1, 2, 10, 0).area)
        assertEquals(12, Rectangle(1, 2, 3, 4).area)
    }

    @Test
    fun test_margin() {
        assertEquals(0, Rectangle(1, 2, 0, 0).margin)
        assertEquals(14, Rectangle(1, 2, 3, 4).margin)
    }
}